open_tms_backend/open_tms_close_ticket/index.js

```diff
@@ -1,4 +1,43 @@
 import { shared } from '@appblocks/node-sdk'
+import { Ticket, TicketRevision, TicketActivity } from './models'
+
+const validateProcessPath = (currentStage, nextStage, assignee) => {
+  const processPath = {
+    open: ['in_progress'],
+    in_progress: ['resolved', 'closed'],
+    resolved: ['closed'],
+  }
+
+  if (processPath[currentStage] && processPath[currentStage].includes(nextStage)) {
+    if (nextStage === 'closed' && assignee !== currentAssignee) {
+      return false
+    }
+    return true
+  }
+  return false
+}
+
+const closeTicket = async (ticketId) => {
+  const latestRevision = await TicketRevision.findOne({
+    where: { ticketId },
+    order: [['id', 'DESC']],
+  })
+
+  if (validateProcessPath(latestRevision.currentStage, 'ticket_closed', latestRevision.assignee)) {
+    const ticket = await Ticket.findByPk(ticketId)
+    ticket.status = 'closed'
+    await ticket.save()
+
+    const activity = await TicketActivity.create({
+      ticketId,
```

Comment:
TicketActivity doesn't have ticketId, instead its having TicketRevisionId. A ticket will have multiple revisions. Please do check the TicketActivity Model and TicketRevision model to get a gist of this relation.